staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -867,12 +869,22 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 		}
 		trace.Step("Object deleted from database")
 
+		status := http.StatusOK
+		// Return http.StatusAccepted if the resource was not deleted immediately and
+		// user requested cascading deletion by setting OrphanDependents=false.
+		// Note: We want to do this always if resource was not deleted immediately, but
+		// that will break existing clients.
+		// Other cases where resource is not instantly deleted are: namespace deletion
+		// and pod graceful deletion.
+		if !wasDeleted && options.OrphanDependents != nil && *options.OrphanDependents == false {
+			status = http.StatusAccepted
+		}
```

Comment:
@smarterclayton Any recommendation on what is the right file to put that test in?
I see `test/integration/auth/auth_test.go` being abused for status code testing. I can add a test there if there is no better place.